### ISSUE TITLE
WIP: Support bitbake 

### ISSUE THREAD

**Issue by rzr** [lvgl/lvgl#2534]
Hi,
I am creating this ticket to track my progress to support LVGL along Bitbake (Yocto/OE).
I plan to upstream some fixes and build recipes somewhere, and will refer to this issue for cooperation purposes

Here is a SDL2 version running on RPi3 on AllScenariOS:

![](https://forum.ostc-eu.org/uploads/default/optimized/1X/7be1f53c210dde35e06150f5a4ec43a4307f2cce_2_690x388.jpeg)

Relate-to: https://git.ostc-eu.org/OSTC/planning/core-os/-/issues/219

**Comment by kisvegabor** [comment]
Thanks for opening this issue. Let us know if you need help form our side.

**Comment by rzr** [comment]
So far it goes fine thx,
I am also about to publish a simple app that can be used as reference,  and reference image.
It's working with sdl2 backend which is still under review ( https://github.com/lvgl/lvgl/pull/2484 )
I'll update this ticket for anyone who want to use it, feedback will be appreciated.
Then some hints about wayland backend would be welcome, but one problem at a time.

Regards

**Comment by kisvegabor** [comment]
> Then some hints about wayland backend would be welcome, but one problem at a time.

Have you already seen its [README](https://github.com/lvgl/lv_drivers/tree/master/wayland)?

**Comment by kisvegabor** [comment]
Not stale

**Comment by rzr** [comment]
yes I plan to upstream recipes to OE, meanwhile check status at:

https://git.ostc-eu.org/rzr/dialog-lvgl/-/wikis/home

**Comment by k3wals** [comment]
Thanks for doing this. I have a primitive recipe for integrating LVGL in Yocto.

In case it's helpful, here's what I added to weston.ini to hide the panel to run fullscreen

panel-position=none

**Comment by rzr** [comment]
You're welcome, feel free to review:
https://github.com/openembedded/meta-openembedded/pull/447

yes I have been using weston as kiosk for an other project (maybe a simple compositor would be desirable too)

**Comment by rzr** [comment]
It's merged in master branch
https://github.com/openembedded/meta-openembedded/commit/0924d41ac4cf968fc302e4f7e52d969cd3899423

Should I keep this issue open for tracking ? 

Thanks for support anyway

**Comment by rzr** [comment]
Let me crosslink this issue about support in dunfell branches:

https://github.com/openembedded/meta-openembedded/pull/453

Meanwhile it can be imported downstream it worked for me too as tracked at:

https://git.ostc-eu.org/OSTC/planning/core-os/-/issues/233


Now I plan to upstream the png lib that can be build as addon (until next release),
please review and merge:
https://github.com/lvgl/lv_lib_png/pulls/rzr

**Comment by kisvegabor** [comment]
Thank your for your great work. 

It'd be great to tell our users that LVGL is available in bitbake. 

Could you add a new file in the [Get started](https://docs.lvgl.io/master/get-started/index.html) section of our docs?

**Comment by rzr** [comment]
yes I can do that :)

**Comment by kisvegabor** [comment]
Hi,

I'm reviewing to open issue to prepare the release of v8.1. 

Could you summarize the state of this topic?

**Comment by rzr** [comment]
well a pre released snapshot has been integrated in OE and let me close this topic once v8.1 is released and also supported then I'll do the documentation part as promised too

**Comment by kisvegabor** [comment]
Okay, thank you.

**Comment by kisvegabor** [comment]
Hi,

What is the current state of this issue?

**Comment by rzr** [comment]
we can close it it's upstreamed  in oe

**Comment by kisvegabor** [comment]
Thanks you! What is the final place where users can find the recipe?

**Comment by rzr** [comment]
Is the current place:

https://github.com/openembedded/meta-openembedded/tree/master/meta-oe/recipes-graphics/lvgl

More details to be explained at:

https://fosdem.org/2022/schedule/event/lvgl/
